File: dummy_code2/secure_sha256.java
// 4. Java Script - Secure SHA-256 Hashing
import java.security.*;

public class SecureSHA256 {
    public static void main(String[] args) throws Exception {
        String data = "password123";
        MessageDigest sha256 = MessageDigest.getInstance("SHA-256");
        byte[] hash = sha256.digest(data.getBytes());
        System.out.println("SHA-256 Hash: " + bytesToHex(hash));
    }

    private static String bytesToHex(byte[] bytes) {
        StringBuilder sb = new StringBuilder();
        for (byte b : bytes) {
            sb.append(String.format("%02x", b));
        }
        return sb.toString();
    }
}

File: dummy_code2/secure_sha256.py
# 1. Python Script - Secure Hashing (SHA-256)
import hashlib

def secure_sha256_hash(data):
    return hashlib.sha256(data.encode()).hexdigest()

plaintext = "password123"
hashed_value = secure_sha256_hash(plaintext)

print("SHA-256 Hash:", hashed_value)